src/AvailableCommandsPacket.php

```diff
@@ -152,6 +154,21 @@ protected function decodePayload(PacketSerializer $in) : void{
 			$this->softEnums[] = $this->getSoftEnum($in);
 		}
 
+		// Add soft enums to command data
+		foreach($this->commandData as $datum){
+			foreach($datum->getOverloads() as $overload){
+				foreach($overload as $parameter){
+					if(($parameter->paramType & self::ARG_FLAG_SOFT_ENUM) !== 0){
+						$index = $parameter->paramType & 0xffff;
+						$parameter->enum = $this->softEnums[$index] ?? null;
+						if($parameter->enum === null){
+							throw new PacketDecodeException("deserializing $datum->name parameter $parameter->paramName: expected soft enum at $index, but got none");
+						}
+					}
+				}
+			}
+		}
```

Comment:
Please extract this logic into a separate function, and document why it's needed (because the soft enums are encoded after the command data that want to reference them).

Reply:
Done